src/app/core/services/issue.service.ts

```diff
@@ -144,8 +144,24 @@ export class IssueService {
           return this.createIssueModel(response);
         }),
         catchError((err) => {
-          this.logger.error('IssueService: ', err); // Log full details of error first
-          return throwError(err.response.data.message); // More readable error message
+          return this.parseUpdateIssueResponseError(err);
+        })
+      );
+  }
+
+  /**
+   * Updates an issue without attempting to create an issue model. Used when we want to treat
+   * updateIssue as an atomic operation that only performs an API call.
+   * @param issue current issue model
+   * @returns GitHubIssue from the API request
+   */
+  updateIssuePure(issue: Issue): Observable<GithubIssue> {
```

Comment:
Seems like this is duplicating the first part of the `updateIssue` method. What do you think about renaming this method as `updateGithubIssue` and then refactoring `updateIssue` to something like
```javascript
updateIssue(issue: Issue): Observable<Issue> {
  return this.updateGithubIssue
    .pipe(
      map((response: GithubIssue) => {
        response.comments = issue.githubComments;
        return this.createIssueModel(response);
      })
    );
}
```

Reply:
Yup! That makes sense, didn't really know what to name it haha, but will do!